MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs

```diff
@@ -218,6 +235,32 @@ private enum Codec
             Decoder
         }
 
+        private IEnumerable<string> GetHwaccelTypes()
+        {
+            string output = null;
+            try
+            {
+                output = GetProcessOutput(_encoderPath, "-hwaccels");
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error detecting available hwaccel types");
+            }
+
+            if (string.IsNullOrWhiteSpace(output))
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var found = output.Split(new char[] {'\r','\n'},StringSplitOptions.RemoveEmptyEntries).Distinct().ToList();
+
+            found.RemoveAt(0);
```

Comment:
Is `.Distinct()` guaranteed to maintain the order?.. If it's not this is a very nasty potential bug...

Reply:
No. Sometimes ffmpeg lists duplicate hwaccel types in `-hwaccels`.

Reply:
This means you might be removing a wrong line.